src/components/Chart/Chart.jsx

```diff
@@ -166,15 +165,19 @@ const Chart = ({ digitalChannelsEnabled = false }) => {
     const dataSelector = useLazyInitializedRef(dataSelectorInitialiser).current;
 
     const { sampleFreq } = useSelector(dataLoggerState);
-    const digitalChannelsWindowLimit = 3e11 / sampleFreq;
-
-    let numberOfBits =
-        windowDuration <= digitalChannelsWindowLimit && showDigitalChannels
-            ? nbDigitalChannels
-            : 0;
-    if (!bits) {
-        numberOfBits = 0;
-    }
+
+    const digitalChannelsWindowLimit = 3e12 / sampleFreq;
+    const zoomedOutTooFarForDigitalChannels =
+        windowDuration > digitalChannelsWindowLimit;
+
+    const digitalChannelsToDisplay = digitalChannels
+        .map((isVisible, channelNumber) => (isVisible ? channelNumber : null))
```

Comment:
[very minor] You could add a prior filter operation here to prevent the conditional within the map.

Reply:
I am not sure whether I understand you correctly. Do you mean I could pull up the `filter` that currently comes below, like the following?
```js
digitalChannels
  .filter(isVisible => isVisible)
  .map((_, channelNumber) => channelNumber)
```

That would not work, since the index of the array is used as `channelNumber` and by filtering first we change that:
```js
const digitalChannels = [false, true, false, true]

digitalChannels
  .filter(isVisible => isVisible)
  .map((_, channelNumber) => channelNumber) 
// => [0, 1]

digitalChannels
  .map((isVisible, channelNumber) => (isVisible ? channelNumber : null))
  .filter(channelNumber => channelNumber != null);
// => [1,3]
```

What I could do though but have not thought about before would be to use flatMap. Not sure whether this might be too cryptic:
```js
digitalChannels.flatMap((isVisible, channelNumber) => isVisible ? channelNumber : [])
```

Reply:
You are right, for some reason I thought I deconstruction in the arguments list, but I see now that channelNumber is the array index, so never mind my comment :)